Lookup resource names in additional answer segment

dave-shawley opened the pull request:
This PR looks up the name from each answer in the additional records segment and returns any address records that it finds there.  If the name does not appear in the additional records, then it is returned as-is (which was the previous behavior).

dave-shawley committed:
tests: Move answer generation into a method.

This makes future changes a little easier as we need to manipulate
the response to exercise branches.

dave-shawley committed:
Look up names in the additional records section of the answer.

pianoman19372 commented:
Your changes look good to me but this library could use some minor additional refactoring:
- dnspython is not pinned in setup.py (should we pin it?)
- theres no testing instructions in the project.

dave-shawley committed:
Pinned dnspython.

We really want at least version 1.12.0 since `options rotate` was not
supported previously.

dave-shawley committed:
Add testing steps to readme.

gmr commented:
@pianoman19372 you should not pin a libraries dependencies unless it's a very _wide_ pin in which case it'd make for a fairly useless pin.
